rfcs/0000-rlp-status.md

```diff
@@ -0,0 +1,162 @@
+
+## Stage 3
+**A final health check of services refines the status**
+
+The final stage will bring a greater degree of accuracy, thanks for a final process that will check the healthiness and 
+configuration version the Kuadrant services currently enforces.
+
+   ![](0000-rlp-status-assets/rlp_status_3.png)
+
+States rationale:
+
+* `Reconciled`: This state will precede the "Health check" process  graphed as `Service Probe` event.
+* `Enforced`: After a successful response of the `Service Probe`, this states communicates the policy is finally enforced.
+This is the final top-level condition.
+* `Failed`: Now this state could also be set after encountering errors in the `Service Probe` check.
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The stages before mentioned, will follow the [Kubernetes guidelines](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) 
+regarding the Status object definition.
+
+
+**Conditions**
+
+| Type       | Status | Reason                    | Message                                                                     | Top Level |
+|------------|--------|---------------------------|-----------------------------------------------------------------------------|-----------|
+| Created    | True   | "PolicyCreated"           | "KuadrantPolicy created"                                                    | No        |
+| Updated    | True   | "PolicyUpdated"           | "KuadrantPolicy has been updated"                                           | No        |
+| Applied    | True   | "PolicyApplied"           | "KuadrantPolicy has been successfully applied                               | Yes       |
+|            | False  | "PolicyNotApplied"        | "KuadrantPolicy is invalid"                                                 |           |
+| Reconciled | True   | "PolicyReconciled"        | "KuadrantPolicy has been successfully reconciled"                           | Yes       |
+|            | False  | "PolicyNotReconciled"     | "KuadrantPolicy failed reconciliation"                                      |           |
+| Enforced   | True   | "PolicyEnforced"          | "KuadrantPolicy has been successfully enforced"                             | Yes       |
+|            | False  | "PolicyPartiallyEnforced" | "KuadrantPolicy has encountered some issues and has been partially applied" |           |
+|            | False  | "PolicyNotEnforced"       | "KuadrantPolicy has encountered an error and can't be applied"              |           |
+
+
+A simplified version and more aligned with the Kubernetes objects implementation could be represented as the following.
+
+**Conditions**
+
+All conditions are top-level.
+
+| Type        | Status | Reason                      | Message                                                                     |
```

Comment:
I think less types would help and it is a good idea to align with K8s or as mentioned above the gateway conditions